File: stringtointegeraoti/sti.py
class Solution:
    def myAtoi(self, str: str) -> int:
        s = str.strip()
        if len(s)==0:
            return 0
        sign,i,n = 1,0,0
        if s[0] == '-':
            sign = -1
            i += 1
        if s[0] == '+':
            i += 1
        while i<len(s):
            if s[i].isdigit() == True:
                n = n*10 + ord(s[i])-ord('0')
                i += 1
            else:
                break
        return sign*min(2**31-1,n) if sign>0 else sign*min(2**31,n)
                
        

File: stringtointegeraoti/sti.java
class Solution {
    public int myAtoi(String str) {
        str = str.trim();
        if(str.length()==0){
            return 0;
        }
        int n = 0;
        int i = 0;
        boolean sign = true;
        if(str.charAt(0) == '-'){
            sign = false;
            i ++;
        }
        if(str.charAt(0) == '+'){
            i ++;
        }
        while(i<str.length()){
            if(Character.isDigit(str.charAt(i))){
                int cur = str.charAt(i)-'0';
                if(n>(Integer.MAX_VALUE-cur)/10){
                    return sign?Integer.MAX_VALUE:Integer.MIN_VALUE;
                }
                n = n*10 + cur;
                i++;
            }
            else{
                break;
            }
        }
        return sign?n:(-1)*n;
    }
}